### ISSUE TITLE
No longer able to launch spotify-tui

### ISSUE THREAD

**Issue by botsunny** [Rigellute/spotify-tui#1043]
OS: Arch Linux
Kernel: 6.0.7-arch1-1
A week ago, I was no longer able to launch spotify-tui by typing `spt` in the kernel. The following error is returned:
```
spt: error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or directory
```
I believe this coincides with the latest update of `openssl` in the Arch repositories, which was recently updated from version 1.1 to 3.0.  Then again, I am unsure about this due to inexperience.

**Comment by ignassew** [comment]
Looks like spotify-tui is no longer maintained, so I doubt the root cause (using openssl 1.1.1) will be fixed anytime soon. You can still install the old version of openssl on arch: `pacman -S openssl-1.1`

**Comment by sewnie** [comment]
No. you just recompile spotify-tui. all AUR packages that depend or are linked to a previous version of OpenSSL must be recompiled to be linked to the new version.

...Unless a dependency crate of spotify-tui needs that specific version of openssl.

**Comment by ignassew** [comment]
> No. you just recompile spotify-tui. all AUR packages that depend or are linked to a previous version of OpenSSL must be recompiled to be linked to the new version.
> 
> ...Unless a dependency crate of spotify-tui needs that specific version of openssl.

Indeed, openssl-sys v0.9.58 (released almost 3 years ago), doesn't support openssl v3.

**Comment by sewnie** [comment]
#1041

**Comment by brookesy66** [comment]
> No. you just recompile spotify-tui. all AUR packages that depend or are linked to a previous version of OpenSSL must be recompiled to be linked to the new version.
> 
> ...Unless a dependency crate of spotify-tui needs that specific version of openssl.

Running `cargo update` then recompiling fixed it for me on Arch Linux as per #1041 as wael444 has linked
